Kitodo-DataAccess/src/main/java/org/kitodo/dataaccess/storage/memory/MemoryNode.java

```diff
@@ -0,0 +1,1096 @@
+                        ObjectType currentCondition = i.next();
+                        if (((AccessibleObject) candidate).matches(currentCondition)) {
+                            i.remove();
+                        }
+                    }
+                } else {
+                    conditions.remove(candidate);
+                }
+            }
+            if (!conditions.isEmpty()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @SuppressWarnings("unchecked") // this is always instanceof Node
+    @Override
+    public <T extends Node> T put(String relation, ObjectType object) {
+        if (relation.equals(RDF.ABOUT)) {
+            throw new IllegalArgumentException(
+                    "Forbidden to put http://www.w3.org/1999/02/22-rdf-syntax-ns#about entries. Use new MemoryNamedNode(identityURL [, data]); instead.");
+        }
+        if (edges.containsKey(relation)) {
+            edges.get(relation).add(object);
+        } else {
+            edges.put(relation, new LinkedList<>(Arrays.asList(new ObjectType[] {object })));
+        }
+        return (T) this;
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @SuppressWarnings("unchecked") // this is always instanceof Node
+    @Override
+    public <T extends Node> T put(String relation, String object) {
+        this.put(relation, MemoryLiteral.createLeaf(object, null));
+        return (T) this;
+    }
+
+    /**
+     * {@inheritDoc}<!-- . -->
+     */
+    @Override
+    public void putAll(String relation, Collection<? extends ObjectType> objects) {
+        if (relation.equals(RDF.ABOUT)) {
+            throw new IllegalArgumentException("Forbidden to put rdf:about entries. Use MemoryNamedNode instead.");
+        }
+        if (objects.isEmpty()) {
+            return;
+        }
+        if (edges.containsKey(relation)) {
+            edges.get(relation).addAll(objects);
+        } else {
+            edges.put(relation, new HashSet<ObjectType>(objects));
+        }
+    }
+
+    /**
+     * Returns the first and last numeric index used in this node to reference
+     * numeric elements. Returns {@code null} if no numeric indexes are in use
+     * at all.
+     *
+     * @return first and last numeric index
+     */
+    private Indices range() {
+        Long last = null;
+        Long first = null;
+        for (Entry<String, Collection<ObjectType>> a : edges.entrySet()) {
+            Long value = RDF.sequenceNumberOf(a.getKey());
+            if (value != null) {
+                if ((first == null) || (value < first)) {
+                    first = value;
+                }
+                if ((last == null) || (value > last)) {
+                    last = value;
+                }
+            }
+        }
+        return (first == null) || (last == null) ? null : new Indices(first, last);
```

Comment:
Returning a `null` value should be avoided. Is a `new Indeces()` not a better solution?